File: examples/nodejs/hello.js
#!/usr/bin/env node
/**
 * Hello World CLI application in Node.js
 * Demonstrates command-line interface, JSON output, and internationalization
 * @author Development Team
 * @version 1.0.0
 */

const { Command } = require('commander');
const chalk = require('chalk');

/**
 * Available greetings in different languages
 * @type {Object<string, string>}
 */
const GREETINGS = {
  en: 'Hello, {}!',
  es: '¡Hola, {}!',
  fr: 'Bonjour, {}!',
  de: 'Hallo, {}!',
  it: 'Ciao, {}!',
  pt: 'Olá, {}!',
  ru: 'Привет, {}!',
  ja: 'こんにちは、{}！',
  zh: '你好，{}！'
};

/**
 * Service class for handling greetings
 */
class GreetingService {
  constructor() {
    this.greetings = GREETINGS;
  }

  /**
   * Generate a greeting message
   * @param {string} name - Name to greet
   * @param {string} language - Language code
   * @returns {string} Greeting message
   */
  greet(name, language = 'en') {
    const template = this.greetings[language] || this.greetings.en;
    return template.replace('{}', name);
  }

  /**
   * Get list of available languages
   * @returns {string[]} Array of language codes
   */
  getAvailableLanguages() {
    return Object.keys(this.greetings);
  }

  /**
   * Create structured greeting data
   * @param {string} name - Name to greet
   * @param {string} language - Language code
   * @returns {Object} Greeting data object
   */
  createGreetingData(name, language = 'en') {
    return {
      message: this.greet(name, language),
      name,
      language,
      timestamp: new Date().toISOString(),
      server: 'Node.js/Express'
    };
  }

  /**
   * Validate if a language is supported
   * @param {string} language - Language code to validate
   * @returns {boolean} True if language is supported
   */
  isLanguageSupported(language) {
    return language in this.greetings;
  }
}

/**
 * Main CLI function
 */
function main() {
  const program = new Command();
  const service = new GreetingService();

  program
    .name('hello')
    .description('Hello World CLI application in Node.js')
    .version('1.0.0');

  program
    .option('-n, --name <name>', 'name to greet', 'World')
    .option('-f, --format <format>', 'output format (text, json)', 'text')
    .option('-l, --language <language>', 'language for greeting', 'en')
    .option('--list-languages', 'list available languages')
    .action((options) => {
      try {
        // Handle list languages
        if (options.listLanguages) {
          const languages = service.getAvailableLanguages();
          
          if (options.format === 'json') {
            console.log(JSON.stringify({ languages }, null, 2));
          } else {
            console.log(chalk.blue('Available languages:'));
            languages.sort().forEach(lang => {
              console.log(`  ${lang}`);
            });
          }
          return;
        }

        // Validate format
        if (!['text', 'json'].includes(options.format)) {
          console.error(chalk.red(`Error: Invalid format '${options.format}'. Use 'text' or 'json'`));
          process.exit(1);
        }

        // Validate language (warn but continue with English if invalid)
        if (!service.isLanguageSupported(options.language)) {
          console.warn(chalk.yellow(`Warning: Language '${options.language}' not supported, using English`));
          options.language = 'en';
        }

        // Generate and output greeting
        if (options.format === 'json') {
          const data = service.createGreetingData(options.name, options.language);
          console.log(JSON.stringify(data, null, 2));
        } else {
          const message = service.greet(options.name, options.language);
          console.log(chalk.green(message));
        }

      } catch (error) {
        console.error(chalk.red('Error:'), error.message);
        process.exit(1);
      }
    });

  program.parse();
}

// Export for testing
module.exports = {
  GreetingService,
  GREETINGS
};

// Run main function if script is executed directly
if (require.main === module) {
  main();
}

File: examples/python/hello.py
#!/usr/bin/env python3
"""
Hello World application in Python
Demonstrates CLI interface, JSON output, and internationalization
"""

import json
import click
from datetime import datetime
from typing import Dict, Optional


def get_greetings() -> Dict[str, str]:
    """Get available greetings in different languages"""
    return {
        "en": "Hello, {}!",
        "es": "¡Hola, {}!",
        "fr": "Bonjour, {}!",
        "de": "Hallo, {}!",
        "it": "Ciao, {}!",
        "pt": "Olá, {}!",
        "ru": "Привет, {}!",
        "ja": "こんにちは、{}！",
        "zh": "你好，{}！",
    }


class GreetingService:
    """Service class for handling greetings"""
    
    def __init__(self):
        self.greetings = get_greetings()
    
    def greet(self, name: str, language: str = "en") -> str:
        """Generate a greeting message"""
        template = self.greetings.get(language, self.greetings["en"])
        return template.format(name)
    
    def get_available_languages(self) -> list:
        """Get list of available languages"""
        return list(self.greetings.keys())
    
    def create_greeting_data(self, name: str, language: str = "en") -> dict:
        """Create structured greeting data"""
        return {
            "message": self.greet(name, language),
            "name": name,
            "language": language,
            "timestamp": datetime.utcnow().isoformat() + "Z",
            "server": "Python/FastAPI"
        }


@click.command()
@click.option("--name", "-n", default="World", help="Name to greet")
@click.option("--format", "-f", default="text", 
              type=click.Choice(["text", "json"]), help="Output format")
@click.option("--language", "-l", default="en", help="Language for greeting")
@click.option("--list-languages", is_flag=True, help="List available languages")
def main(name: str, format: str, language: str, list_languages: bool):
    """Hello World CLI application"""
    
    service = GreetingService()
    
    if list_languages:
        languages = service.get_available_languages()
        if format == "json":
            click.echo(json.dumps({"languages": languages}, indent=2))
        else:
            click.echo("Available languages:")
            for lang in languages:
                click.echo(f"  {lang}")
        return
    
    if format == "json":
        data = service.create_greeting_data(name, language)
        click.echo(json.dumps(data, indent=2))
    else:
        message = service.greet(name, language)
        click.echo(message)


if __name__ == "__main__":
    main()